### PR TITLE
stargate: split x/evm/types/

### PR DESCRIPTION
**Pull request by fedekunze**
- stargate: split proto and base files
- stargate: split app/ + codec/
- stargate: split types/
- stargate: split crypto/
- stargate: split x/evm/types/

### CODE REVIEW

File: proto/ethermint/evm/v1alpha1/genesis.proto [hunk 1]
```diff
@@ -30,13 +30,8 @@ message GenesisState {
...
-  // balance defines the available evm tokens for the account
-  string balance = 2 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.nullable) = false
-  ];
   // code defines the hex bytes of the account code.
-  string code = 3;
+  string code = 2;
   // storage defines the set of state key values for the account.
   repeated State storage = 4 [
```
Comment (noot) [inline proto/ethermint/evm/v1alpha1/genesis.proto]: no `3` anymore?

File: x/evm/types/state_transition.go [hunk 2]
```diff
@@ -79,13 +80,13 @@ func (st StateTransition) newEVM(
...
 ) *vm.EVM {
-	// Create contexts for evm
+	// Create context for evm
 
 	blockCtx := vm.BlockContext{
 		CanTransfer: core.CanTransfer,
 		Transfer:    core.Transfer,
 		GetHash:     GetHashFn(ctx, csdb),
-		Coinbase:    common.Address{}, // there's no beneficiary since we're not mining
+		Coinbase:    common.Address{}, // there's no benefitiary since we're not mining
```
Comment (noot) [inline x/evm/types/state_transition.go]: I think the correct spelling is `beneficiary`

File: x/evm/types/state_transition_test.go [hunk 3]
```diff
@@ -23,51 +25,60 @@ func (suite *StateDBTestSuite) TestGetHashFn() {
...
-						ChainID:        "ethermint-1",
-						Height:         1,
-						ValidatorsHash: []byte("val_hash"),
-					},
-				)
-			},
-			false,
-		},
+		// {
+		// 	"valid hash, case 1",
```
Comment (noot) [inline x/evm/types/state_transition_test.go:29]: why are these commented out?

File: x/evm/types/state_transition_test.go [hunk 4]
```diff
@@ -196,6 +210,24 @@ func (suite *StateDBTestSuite) TestTransitionDb() {
 			},
 			false,
 		},
+		{
+			"failed to Finalize",
```
Comment (noot) [inline x/evm/types/state_transition_test.go:214]: what case is this testing?

File: proto/ethermint/evm/v1alpha1/genesis.proto (hunk 1 above)
Comment (fedekunze) [reply proto/ethermint/evm/v1alpha1/genesis.proto]: good catch

File: x/evm/types/state_transition_test.go (hunk 3 above)
Comment (fedekunze) [reply x/evm/types/state_transition_test.go:29]: idk prob some broken tests, I can uncomment/fix them

File: x/evm/types/state_transition_test.go (hunk 4 above)
Comment (fedekunze) [reply x/evm/types/state_transition_test.go:214]: when the finalize function fails

File: x/evm/types/chain_config.go [hunk 5]
```diff
@@ -4,46 +4,13 @@ import (
...
 
-// ChainConfig defines the Ethereum ChainConfig parameters using sdk.Int values instead of big.Int.
-//
-// NOTE 1: Since empty/uninitialized Ints (i.e with a nil big.Int value) are parsed to zero, we need to manually
-// specify that negative Int values will be considered as nil. See getBlockValue for reference.
-//
-// NOTE 2: This type is not a configurable Param since the SDK does not allow for validation against
-// a previous stored parameter values or the current block height (retrieved from context). If you
-// want to update the config values, use an software upgrade procedure.
-type ChainConfig struct {
```
Comment (araskachoi) [inline x/evm/types/chain_config.go:24]: chainconfig is now defined in the protobuf go file? how come some types are defined twice, once in the ethermint library and again in the pb files while others (like chainconfig) only gets defined once?

File: x/evm/types/events.go [hunk 6]
```diff
@@ -2,7 +2,6 @@ package types
 
 // Evm module events
 const (
-	EventTypeEthermint  = TypeMsgEthermint
```
Comment (araskachoi) [inline x/evm/types/events.go:5]: why no more ethermint event?

Comment (fedekunze) [reply x/evm/types/events.go:5]: we don't have msg ethermint anymore

File: x/evm/types/msg.go [hunk 7]
```diff
@@ -422,7 +319,13 @@ func (msg *MsgEthereumTx) VerifySig(chainID *big.Int) (ethcmn.Address, error) {
 		return ethcmn.Address{}, err
 	}
 
-	msg.from.Store(sigCache{signer: signer, from: sender})
+	msg.From = &SigCache{
+		Signer: &EIP155Signer{
+			chainId:    chainID.Bytes(),
+			chainIdMul: new(big.Int).Mul(chainID, big.NewInt(2)).Bytes(),
```
Comment (noot) [inline x/evm/types/msg.go:325]: what is this multiplier?

**Review by noot — APPROVED** [review]
pls either remove or fix commented out tests, everything else looks good!

**Pull request closed without merging**